node/src/actors/chain_manager/handlers.rs

```diff
@@ -399,6 +399,10 @@ impl Handler<AddBlocks> for ChainManager {
                         log::debug!("TargetNotReached: superblock target #{} not reached, requesting more blocks. ({} processed blocks)",
                             sync_target.superblock.checkpoint, num_processed_blocks);
                         self.request_blocks_batch(ctx);
+
+                        // Copy current chain state into previous chain state, and persist it
+                        self.move_chain_state_forward(0);
+                        self.persist_chain_state(0, ctx);
```

Comment:
This `0` should be the superblock index, but here we do not use the superblock index, we use the sync blocks batch. How about we change it into an option, and use `None` here but `Some(_)` everywhere else?

This way in theory we can distinguish if the chain state is valid because it is a consolidated block, or it may be invalid because we received it while syncing.